controller/linodemachine_controller.go

```diff
@@ -378,6 +391,66 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 	return linodeInstance, nil
 }
 
+func (r *LinodeMachineReconciler) configureDisks(
+	ctx context.Context,
+	logger logr.Logger,
+	machineScope *scope.MachineScope,
+	linodeInstanceID int,
+) error {
+	// get the default instance config
+	configs, err := machineScope.LinodeClient.ListInstanceConfigs(ctx, linodeInstanceID, &linodego.ListOptions{})
+	if err != nil || len(configs) == 0 {
+		logger.Error(err, "Failed to list instance configs")
+
+		return err
+	}
+	instanceConfig := configs[0]
+
+	// carve out space for the etcd disk
+	rootDiskID := instanceConfig.Devices.SDA.DiskID
+	rootDisk, err := machineScope.LinodeClient.GetInstanceDisk(ctx, linodeInstanceID, rootDiskID)
+	if err != nil {
+		logger.Error(err, "Failed to get root disk for instance")
+
+		return err
+	}
+	diskSize := rootDisk.Size - defaultEtcdDiskSize
+	if err = machineScope.LinodeClient.ResizeInstanceDisk(ctx, linodeInstanceID, rootDiskID, diskSize); err != nil {
+		logger.Error(err, "Failed to resize root disk")
+
+		return err
+	}
+
+	// create the etcd disk
+	etcdDisk, err := machineScope.LinodeClient.CreateInstanceDisk(
+		ctx,
+		linodeInstanceID,
+		linodego.InstanceDiskCreateOptions{
+			Label:      "etcd-data",
+			Size:       defaultEtcdDiskSize,
+			Filesystem: string(linodego.FilesystemExt4),
+		},
+	)
+	if err != nil {
+		logger.Error(err, "Failed to create etcd disk")
+
+		return err
+	}
+
+	// attach etcd disk to sdc
+	instanceConfig.Devices.SDC = &linodego.InstanceConfigDevice{DiskID: etcdDisk.ID}
```

Comment:
these should be statitically allocated during config profile right, and never detached?

Reply:
yes, I also checked to make sure this stays in the event of an instance reboot.